src/containers/NoMatch/index.tsx

```diff
@@ -40,19 +40,20 @@ const NoMatch = ({
   }, [...hints, title, track])
 
   const notFound = title.includes('not_found')
-  const hintMsg = hints.map((hint) => (
-    <div className="hint" key={hint}>
-      {t(hint as any, values)}
-    </div>
-  ))
+
+  // Determine which hint message to display based on the presence of publicKey
+  const hintMsg = values.connection?.server?.publicKey
+    ? t('server_ledgers_hint_with_key', values)
+    : t('server_ledgers_hint', values)
+
   const derivedWarning = warning ?? (notFound && t('not_found'))
 
   return (
     <div className="no-match">
       <Helmet title={t(title as any)} />
       {isError && <div className="uh-oh">{t('uh_oh')}</div>}
       <div className="title">{t(title as any, values)}</div>
-      {hintMsg}
+      {hintMsg && <div className="hint">{hintMsg}</div>}
```

Comment:
I'm not sure why we need to change here. Would `hintMsg` ever be `null`? Also, are we not using `hints` array here at all?

Reply:
Correct, at present, the _hints_ array isn't directly used within the JSX rendering logic of the component.
Instead, it's utilized within the _useEffect_ hook to concatenate its values into the _description_ parameter of the _track_ function.